jax/_src/lax/parallel.py

```diff
@@ -980,6 +963,7 @@ def _axis_index_bind(*, axis_name):
 axis_index_p.def_custom_bind(_axis_index_bind)
 
 def _process_axis_index(self, frame):
+  assert frame.size is not None
```

Comment:
When can that happen?

Reply:
This is related to what I mentioned in the PR message: namely, that in #5492 I changed `batching.batch` not to compute the axis size and instead take it as an argument, which worked with all call sites _except_ `pxla.vtile`, where it could be a `None`. The result was a `None` in the core axis env, and it was caught pretty late (only when it was consumed by a rule like this one). This particular assert doesn't catch it much earlier but the error message is slightly cleaner.